base/strings/faces.jl

```diff
@@ -0,0 +1,779 @@
+    set by the Symbol, one of `:straight`, `:double`, `:curly`, `:dotted`,
+    or `:dashed`.\\
+  - a `Tuple{SimpleColor, Symbol}`: The text should be underlined in the specified
+    SimpleColor, and using the style specified by the Symbol, as before.
+- `strikethrough` (a `Bool`): Whether the text should be struck through.
+- `inverse` (a `Bool`): Whether the foreground and background colors should be
+  inverted.
+- `inherit` (a `Vector{Symbol}`): Names of faces to inherit from,
+  with earlier faces taking priority. All faces inherit from the `:default` face.
+"""
+struct Face
+    font::Union{Nothing, String}
+    foreground::Union{Nothing, SimpleColor}
+    background::Union{Nothing, SimpleColor}
+    weight::Union{Nothing, Symbol}
+    slant::Union{Nothing, Symbol}
+    underline::Union{Nothing, Bool, SimpleColor,
+                     Tuple{<:Union{Nothing, SimpleColor}, Symbol}}
+    strikethrough::Union{Nothing, Bool}
+    inverse::Union{Nothing, Bool}
+    inherit::Vector{Symbol}
+end
+
+function Face(; font::Union{Nothing, String} = nothing,
+              foreground::Union{Nothing, SimpleColor, Symbol, RGBTuple, UInt32} = nothing,
+              background::Union{Nothing, SimpleColor, Symbol, RGBTuple, UInt32} = nothing,
+              weight::Union{Nothing, Symbol} = nothing,
+              slant::Union{Nothing, Symbol} = nothing,
+              underline::Union{Nothing, Bool, SimpleColor,
+                               Symbol, RGBTuple, UInt32,
+                               Tuple{<:Union{Nothing, SimpleColor, Symbol, RGBTuple, UInt32}, Symbol}
+                               } = nothing,
+              strikethrough::Union{Nothing, Bool, SimpleColor, Symbol, RGBTuple, UInt32} = nothing,
+              inverse::Union{Nothing, Bool} = nothing,
+              inherit::Union{Symbol, Vector{Symbol}} = Symbol[],
+              _...) # Simply ignore unrecognised keyword arguments.
+    ascolor(::Nothing) = nothing
+    ascolor(c::SimpleColor) = c
+    ascolor(c::Union{Symbol, RGBTuple, UInt32}) = SimpleColor(c)
+    Face(font, ascolor(foreground), ascolor(background), weight, slant,
+         if underline isa Tuple
+             (ascolor(underline[1]), underline[2])
+         elseif underline isa Symbol || underline isa RGBTuple || underline isa UInt32
+             ascolor(underline)
+         else
+             underline
+         end,
+         if strikethrough isa Bool
+             strikethrough
+         else ascolor(strikethrough) end,
+         inverse,
+         if inherit isa Symbol
+             [inherit]
+         else inherit end)
+end
+
+==(a::Face, b::Face) =
+    getfield.(Ref(a), fieldnames(Face)) ==
+    getfield.(Ref(b), fieldnames(Face))
+
+"""
+The default values for all defined faces. [`FACES`](@ref) is initially set to a
+copy of this value. This essentially allows for any modifications to
+[`FACES`](@ref) to be undone.
+"""
+const DEFAULT_FACES = Dict{Symbol, Face}(
```

Comment:
Should this maybe be `_DEFAULT_FACES`? 

It'd be nice to signal that the values in here should never change. Updating the docstring to state that would also help.

Reply:
Hmm, this could be done. Really neither `DEFAULT_FACES` not `FACES` should be directly accessed. I also wonder if `FACES` should be renamed to `CURRENT_FACES` or this changed to a `NamedTuple{(:default, :current)}`.

Reply:
This thought is complicated by the fact that one is a `const` and the other isn't. I'm also wary of just slapping underscores in front of everything non-public API.

Reply:
Going to try a NamedTuple approach.